ODCodeGen/CatOD2_CodeGen/CatOD2.training.cs

```diff
@@ -22,14 +22,14 @@ public partial class CatOD2
      public const int TrainingImageWidth = 800;
      public const int TrainingImageHeight = 600;
 
-     /// <summary>
-     /// Train a new model with the provided dataset.
-     /// </summary>
-     /// <param name="outputModelPath">File path for saving the model. Should be similar to "C:\YourPath\ModelName.mlnet"</param>
-     /// <param name="inputDataFilePath">Path to the data file for training.</param>
-     public static void Train(string outputModelPath, string inputDataFilePath = RetrainFilePath)
+        /// <summary>
+        /// Train a new model with the provided dataset.
+        /// </summary>
+        /// <param name="outputModelPath">File path for saving the model. Should be similar to "C:\YourPath\ModelName.mlnet"</param>
+        /// <param name="inputDataFilePath">Path to the data file for training.</param>
+        public static void Train(string outputModelPath, MLContext? mlContext = null, string inputDataFilePath = RetrainFilePath)
```

Comment:
I like being able to override the mlContext so that I can hook up the logger. It's okay if we change this code for all scenarios and codegen.